ledger/participant-state/kvutils/src/main/scala/com/daml/ledger/participant/state/kvutils/Conversions.scala

```diff
@@ -246,10 +239,13 @@ private[state] object Conversions {
   /** Encodes a [[BlindingInfo]] into protobuf (i.e., [[DamlTransactionBlindingInfo]]).
     * It is consensus-safe because it does so deterministically.
     */
-  def encodeBlindingInfo(blindingInfo: BlindingInfo): DamlTransactionBlindingInfo =
+  def encodeBlindingInfo(
+      blindingInfo: BlindingInfo,
+      divulgedContracts: Map[ContractId, TransactionOuterClass.ContractInstance],
+  ): DamlTransactionBlindingInfo =
     DamlTransactionBlindingInfo.newBuilder
       .addAllDisclosures(encodeDisclosure(blindingInfo.disclosure).asJava)
-      .addAllDivulgences(encodeDivulgence(blindingInfo.divulgence).asJava)
+      .addAllDivulgences(encodeDivulgence(blindingInfo.divulgence, divulgedContracts).asJava)
       .build
 
   def decodeBlindingInfo(blindingInfo: DamlTransactionBlindingInfo): BlindingInfo =
```

Comment:
`com.daml.lf.transaction.BlindingInfo` cannot cannot be easily enriched with `contractInstance` counterpart in the divulgence map. Reasons:
* It is widely used and code impact is significant both in this repo.
* The `com.daml.lf.transaction.BlindingInfo`, as a result of the `Blinding.blind` cannot be populated with the contract instances when built.

Hence the conversion `DamlTransactionBlindingInfo` cannot be made revertible in here. I am not happy about this. The impact is local to KVUtils, but I am wondering whether it is not actually worse to introduce this misalignment than to just leave the divulged contract instances in the initially-proposed `DamlTransactionEntry.divulgedContracts`. What do you think, @miklos-da ?